src/v/redpanda/admin_server.cc

```diff
@@ -1424,6 +1425,119 @@ void config_multi_property_validation(
           "{} requires schema_registry to be enabled in redpanda.yaml", name);
     }
 }
+
+/**
+ * A helper method to map locked cluster properties to their corresponding
+ * topic-level properties
+ * TODO(@NyaliaLui): Define a type mapping and templatize this method because
+ * configuration properties do not have the same type. Also, keep in mind that
+ * some topic_configs are optional and some are tristate.
+ */
+std::optional<uint64_t> topic_property_from_locked_property(
+  const ss::sstring& locked_prop_name,
+  const cluster::topic_configuration& tp_config) {
+    // TODO(@NyaliaLui): Don't forget to include compacted_log_segment_size
+    if (
+      locked_prop_name == "log_segment_size_locked_min"
+      || locked_prop_name == "log_segment_size_locked_max") {
+        return tp_config.properties.segment_size;
+    } else {
+        // TODO(@NyaliaLui): Consider a static_assert and making the method
+        // constexpr
+        vlog(logger.error, "Undefined locked property name");
+        return std::nullopt;
+    }
+}
+
+/**
+ * A helper method to check locked configs in the request. Throws if:
+ * - Requested locked min > locked max
+ * - A pre-existing topic violates a requested locked config
+ */
+void check_locked_configs(
+  const cluster::config_update_request& req,
+  const cluster::topic_table& topic_table) {
+    const std::regex locked_min_regex("^[a-z_]+locked_min$"),
+      locked_max_regex("^[a-z_]+locked_max$");
+    const std::regex locked_bool_regex("^[a-z_]+locked$");
```

Comment:
Please use one line per declaration.

I think it would be far simpler to rely on the `ends_with` member function of `ss::sstring` than use a regex.

Reply:
Excellent suggestion.